### PR TITLE
[kube-prometheus-stack] added support for cert duration customization;

### LINKED ISSUE #551: [kube-prometheus-stack] Support customized certificate duration when using cert-manager as tls certificate provisioner
**Is your feature request related to a problem? Please describe.**
cert-manager support in prometheus operator admission webhook does not support customized duration. It is hardcoded to be 1 year. Moreover, its hardcoded value "8760h" is parsed as 8760h0m0s when applied in k8s, which may cause trouble in automatically managed k8s environment like ArgoCD (it will assume 8760h is different from 8760h0m0s and keep synchronizing, which is unnecessary)

**Describe the solution you'd like**
Support customized certificate duration (e.g. add a field in value file `.Values.prometheusOperator.admissionWebhooks.certManager.duration`). I can create a pull request if I am allowed to do so.

**Describe alternatives you've considered**
Replace 8760h with 8760h0m0s.

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by ethernoy**
Signed-off-by: Ethern Su <ehaprime@gmail.com>

#### What this PR does / why we need it:

This PR added support for customization on cert duration in kube-prometheus-stack helm chart.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #551

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/kube-prometheus-stack/templates/prometheus-operator/certmanager.yaml [hunk 1]
```diff
@@ -18,7 +18,7 @@ metadata:
   namespace: {{ template "kube-prometheus-stack.namespace" . }}
 spec:
   secretName: {{ template "kube-prometheus-stack.fullname" . }}-root-cert
-  duration: 43800h0m0s # 5y
+  duration: {{ .Values.prometheusOperator.admissionWebhooks.certManager.rootCert.duration }}
```
Comment (gkarthiks) [inline charts/kube-prometheus-stack/templates/prometheus-operator/certmanager.yaml:21]: how about we add the template for the duration and if not present make the default as 5y. In this way, we have support for both and folks don't need to edit the default values.yml file until they desired to change the duration.

Does that sound reasonable to you?

**Review by gkarthiks — CHANGES_REQUESTED** [review]

File: charts/kube-prometheus-stack/templates/prometheus-operator/certmanager.yaml (hunk 1 above)
Comment (ethernoy) [reply charts/kube-prometheus-stack/templates/prometheus-operator/certmanager.yaml:21]: Very reasonable, will do it soon

Comment (ethernoy) [reply charts/kube-prometheus-stack/templates/prometheus-operator/certmanager.yaml]: Updated

**Comment by gkarthiks** [comment]
@ethernoy  can you please fix this https://github.com/prometheus-community/helm-charts/runs/4628200438?check_suite_focus=true#step:8:37

**Comment by ethernoy** [comment]
> @ethernoy can you please fix this https://github.com/prometheus-community/helm-charts/runs/4628200438?check_suite_focus=true#step:8:37

@gkarthiks updated and passed the checking

File: charts/kube-prometheus-stack/Chart.yaml [hunk 2]
```diff
@@ -17,7 +17,7 @@ name: kube-prometheus-stack
 sources:
   - https://github.com/prometheus-community/helm-charts
   - https://github.com/prometheus-operator/kube-prometheus
-version: 25.1.0
+version: 25.1.1
```
Comment (monotek) [inline charts/kube-prometheus-stack/Chart.yaml:20]: imho this is not a bugfix but a new feature. therefore minor version should be raised.

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/kube-prometheus-stack/Chart.yaml (hunk 2 above)
Comment (ethernoy) [reply charts/kube-prometheus-stack/Chart.yaml:20]: Sure

**Review by monotek — APPROVED** [review]

**Pull request merged**